packages/teams-js/src/internal/communication.ts

```diff
@@ -562,6 +737,16 @@ function createMessageRequest(func: string, args: any[] | undefined): MessageReq
   };
 }
 
+function createNestedAppAuthRequest(message: string): NestedAuthRequest {
+  return {
+    id: CommunicationPrivate.nextMessageId++,
+    func: 'nestedAppAuthRequest',
+    timestamp: Date.now(),
+    args: [],
+    data: message,
```

Comment:
Might be a dumb question but I will ask anyways: why does data need to be separate property? It seems like it would "fit" more with the existing contract if args contained the message

Reply:
I didn't want to use `args` because I didn't want to overload it. It also reduces the probablity of errors in the HubSDK itself i.e. if HubSDK is updated to pre-process `args` in the future, NAA messages might be miscategorized or throw errors since `args` are meant to be parameters to a callback in the Hub. 

Keeping the `data` in `event.data` would be an organic approach since it is siloed for NAA requests. Specifically, in the message processors since we saw that we have access to the complete `event`, we don't need to change anything but just rely on the existing pattern and get the `data` destructured from the event in the NAA processor.